Sources/MapboxNavigation/NavigationMapView+Annotations.swift

```diff
@@ -0,0 +1,488 @@
+           let railroadCrossingDay =  Bundle.mapboxNavigation.image(named: "RailroadCrossingDay") {
+            try style.addImage(railroadCrossingDay, id: ImageIdentifier.railroadCrossingDay)
+        }
+        
+        if style.image(withId: ImageIdentifier.railroadCrossingNight) == nil,
+           let railroadCrossingNight =  Bundle.mapboxNavigation.image(named: "RailroadCrossingNight") {
+            try style.addImage(railroadCrossingNight, id: ImageIdentifier.railroadCrossingNight)
+        }
+    }
+    
+    func updateIntersectionSignals(with routeProgress: RouteProgress) {
+        guard !routeProgress.routeIsComplete else {
+            removeIntersectionSignals()
+            return
+        }
+        var featureCollection = FeatureCollection(features: [])
+        
+        let stepProgress = routeProgress.currentLegProgress.currentStepProgress
+        let intersectionIndex = stepProgress.intersectionIndex
+        let stepIntersections = stepProgress.intersectionsIncludingUpcomingManeuverIntersection
+        
+        for intersection in stepIntersections?.suffix(from: intersectionIndex) ?? [] {
+            if let feature = signalFeature(from: intersection, styleType: styleType) {
+                featureCollection.features.append(feature)
+            }
+        }
+        
+        let style = mapView.mapboxMap.style
+        
+        do {
+            let sourceIdentifier = NavigationMapView.SourceIdentifier.intersectionSignalSource
+            if style.sourceExists(withId: sourceIdentifier) {
+                try style.updateGeoJSONSource(withId: sourceIdentifier, geoJSON: .featureCollection(featureCollection))
+            } else {
+                var source = GeoJSONSource()
+                source.data = .featureCollection(featureCollection)
+                try style.addSource(source, id: sourceIdentifier)
+            }
+            
+            let layerIdentifier = NavigationMapView.LayerIdentifier.intersectionSignalLayer
+            guard !style.layerExists(withId: layerIdentifier) else { return }
+            
+            var shapeLayer = SymbolLayer(id: layerIdentifier)
+            shapeLayer.source = sourceIdentifier
+            shapeLayer.iconAllowOverlap = .constant(false)
+            shapeLayer.iconImage = .expression(Exp(.get) {
+                "imageName"
+            })
+            
+            let layerPosition = layerPosition(for: layerIdentifier)
+            try style.addPersistentLayer(shapeLayer, layerPosition: layerPosition)
+        } catch {
+            Log.error("Failed to perform operation while adding intersection signals with error: \(error.localizedDescription).",
+                      category: .navigationUI)
+        }
+    }
+    
+    private func signalFeature(from intersection: Intersection, styleType: StyleType) -> Feature? {
+        var properties: JSONObject? = nil
+        let styleTypeString = (styleType == .day) ? "Day" : "Night"
```

Comment:
As long as `updateIntersectionSymbolImages()` is automatically called on style changes, we won’t need to embed the style name inside a feature property. This feature is being displayed by a persistent layer, which I think would have a longer lifespan than the style image anyways.

Reply:
This style type name has been removed from the feature properties, and the style change will trigger the `updateIntersectionSymbolImages(styleType:)` instead